### PR TITLE
docs: Expand on cleanup behaviour for submission handlers

### PR DESCRIPTION
**Pull request by Nessworthy**
Cover use cases for both using a submission handler that returns a promise, and for a one which does not.

Add FAQ entry about submission handler cleanup.

Following submission handling logic from https://github.com/jaredpalmer/formik/blob/main/packages/formik/src/Formik.tsx#L754-L782

### CODE REVIEW

**Comment by Nessworthy** [comment]
Please squash if merged, do not immortalise my formatting mistakes 😂

File: docs/guides/form-submission.md [hunk 1]
```diff
@@ -23,8 +23,10 @@ To submit a form in Formik, you need to somehow fire off the provided `handleSub
 
 ### Submission
 
-- Proceed with running your submission handler (i.e.`onSubmit` or `handleSubmit`)
-- _you call `setSubmitting(false)`_ in your handler to finish the cycle
+- Proceed with running your submission handler (i.e. `onSubmit` or `handleSubmit`)
+- Did the submit handler return a promise?
+  - Yes: Wait until it is resolved or rejected, then set `setSubmitting` to `false`
+  - No: _You call `setSubmitting(false)`_ in your handler to finish the cycle
```
Comment (probablyup) [inline docs/guides/form-submission.md:29]:
```suggestion
  - No: _Call `setSubmitting(false)`_ to finish the cycle
```

File: docs/guides/form-submission.md [hunk 2]
```diff
@@ -55,3 +57,12 @@ Disable whatever is triggering submission if `isSubmitting` is `true`.
 If `isValidating` is `true` and `isSubmitting` is `true`.
 
 </details>
+
+<details>
+<summary>Why does isSubmitting remain true after submission?</summary>
```
Comment (probablyup) [inline docs/guides/form-submission.md:62]:
```suggestion
<summary>Why does `isSubmitting` remain true after submission?</summary>
```

File: docs/guides/form-submission.md [hunk 3]
```diff
@@ -23,8 +23,10 @@ To submit a form in Formik, you need to somehow fire off the provided `handleSub
 
 ### Submission
 
-- Proceed with running your submission handler (i.e.`onSubmit` or `handleSubmit`)
-- _you call `setSubmitting(false)`_ in your handler to finish the cycle
+- Proceed with running your submission handler (i.e. `onSubmit` or `handleSubmit`)
```
Comment (probablyup) [inline docs/guides/form-submission.md:26]:
```suggestion
- Proceed with running the submission handler (i.e. `onSubmit` or `handleSubmit`)
```

**Review by probablyup — APPROVED** [review]
Thanks :)

**Review by jaredpalmer — APPROVED** [review]

**Pull request merged**